plane/fit_plane.py

```diff
@@ -0,0 +1,72 @@
+import numpy as np
+# from plane.fit_plane import FitPlane
+from plane.fit_template import FitTemplate
+import pandas as pd
+
+class FitPlane:
+    def __init__(self, fitplanes, target_centers, template_size, um_per_pixel):
+        self.fitplanes = fitplanes
+        self.real_centers = target_centers # in um
+        self.template_size = template_size
+        self.um_per_pixel = um_per_pixel
+        self.fitplane_centers = self.calc_fitplane_centers() # in um
+        self.distances = self.calc_distances() # in um
+
+    @classmethod
+    def from_aligned_fit_templates(cls, fit_templates_list, target_centers_list, template_size=401, um_per_pixel=2):
+        """
+        Function to calculate/store the params for individual barcodes and combinations of barcodes. 
+
+        :param fit_templates_list: list of barcodes contained in this FitPlane. Each is a FitTemplate object.
+        :param target_centers_list: theoretical positions of each barcode center as defined by photobleach script.
+        :param template_size: square edge length of the template image used for alignment in each FitTemplate, in pixels.
+        :param um_per_pixel: um per pixel in the template image.
+        :returns: Initializes an instance of a FitPlane.
+        """
+        return cls(fit_templates_list, target_centers_list, template_size, um_per_pixel)
+
+    def __len__(self):
+        return len(self.fitplanes)
+    
+    def calc_distances(self):
```

Comment:
Unclear what this function does.
Is it mirroring something from coregister_xz's fit_plane.py?

Reply:
has been removed- it is not useful